src/jobservice/pool/redis_pool.go

```diff
@@ -593,6 +610,37 @@ func (rpc *RedisPoolContext) logJob(job *work.Job, next work.NextMiddlewareFunc)
 	return next()
 }
 
+//Ping the redis server
+func (gcwp *GoCraftWorkPool) ping() error {
+	conn := gcwp.redisPool.Get()
+	defer conn.Close()
+
+	//First try
+	if _, err := conn.Do("ping"); err == nil {
+		return nil
+	}
+
+	count := 1
+	timer := time.NewTimer(time.Duration(count) * time.Second)
+	defer timer.Stop()
+
+	var err error
+	for {
+		select {
+		case <-timer.C:
```

Comment:
this may be trivial, but there's only one channel to be listened, by do we need to use `select`?
And why not use time.Sleep?